.github/workflows/main.yml

```diff
@@ -70,13 +70,22 @@ jobs:
         if: ${{ matrix.python-version == 2.7 }}
         run: tox --workdir .github/workflows/.tox --recreate -e py38-covcombine
 
-      - name: Upload coverage reports to Codecov
+      - name: Upload the Python 2.7 coverage report to Codecov
+        if: ${{ matrix.python-version == 2.7 }}
+        uses: codecov/codecov-action@v3
+        with:
+          directory: .github/workflows/.tox/py27-cov/log
+          env_vars: OS,PYTHON
+          fail_ci_if_error: true
+          flags: Python-2.7
+          verbose: true
+
+      - name: Upload the Python 3.8 coverage report to Codecov
         if: ${{ matrix.python-version == 2.7 }}
```

Comment:
Passing the coverage.xml from each and all python interpreter matrix jobs to a special upload job may be possible, but then they would all have the same flags parameter and I think we want to see the different Python version as flags. So, no, it's not a good idea. Each coverage upload is very fast anyways any even if it were possible, it would not be worth micro-optimizing this last tiny bit, e.g. using a custom uploader task which could one flag per coverage file, and as nobody else would use it, we would create technical dept if we'd implement that.

Reply:
> Passing the coverage.xml from each and all python interpreter matrix jobs to a special upload job may be possible

I was thinking about uploading the coverage.xml on all the different jobs, independently, rather than a separate job that waits on all coverage.xmls